### DISCUSSION TITLE
Vercel Secrets Management

### DISCUSSION THREAD

**Discussion by kennetpostigo** [vercel/vercel#4203] [General] [answered]
Over the course of using Zeit since it's pretty much been a business theres some pretty cool stuff you can do that maybe hasn't been explored. One of them we've been hesitant to talk about because we kind of weren't sure if talking about it would make Zeit want to charge us for.

We built a home grown auth service around `now secrets` and rotating api keys programmatically. It worked pretty well and has been awesome not needing to pay for something like Auth0 since we don't use most of it's features and also we're a tiny company that is bootstrapped by our little team and pinching pennies.

However we pretty much had to ditch it as we restructured our setup and folded most of our projects/services into one service that namespaces those services in `/api/service` now mostly to decrease maintenance and integration overhead. Simply changing the boundaries of what separated our services made the whole team more productive.

To get to the point once we finished restructuring we hit the secrets 4kb limit since our keys were in one project. It would be cool for Zeit to maybe explore this to make a less limited offering integrated into the platform . 

Since the change we moved our secrets and data protection management by setting up Hashicorps Vault in Digital Ocean and creating a client api above the nodejs client that we use in our now lambdas. It's working well but now we work above 2 platforms instead of just `now`. It's been working out great but would've been cool to be able to store more secrets on top of now. 

There are a few services that offer secrets management pretty well that don't limit size and we've thought about and will probably migrate to later in the year once we create tooling to migrate off of `now` but thought we might as well put it in here so it's something ya'll can explore. 

Up Pro offers pretty nice secrets management without limiting size/usage. The downside is that secrets injection happen at deployment time in Up. We have a pretty nice scheduler so we can probably schedule out deployments and upgrade new ones as keys rotate so it's not a problem for us.

Anyway, just some ideas on that front

**Comment by coetry** [comment, marked as answer]
Hey @kennetpostigo ! That sounds really cool, but we offer secrets management mainly as a way to integrate with your web apps in a painless manner. The 4kb limit is an upstream limitation and we aren't interested in exploring secrets management outside the scope of web app :smile: 

>  we kind of weren't sure if talking about it would make Zeit want to charge us for.

We (Vercel) :wink: aren't interested in charging you for use of our products within the established limits. Enjoy :rocket:
